lib/m2ts/probe.js

```diff
@@ -113,6 +114,28 @@ var parsePesType = function(packet, programMapTable) {
   }
 };
 
+// given bytes from an elementary stream starting at a PES header, try to find the ADTS
+// header offset
+var getAdtsHeaderOffset = function(packet) {
+  // must ensure we are past stuffing bytes by checking the adaptation field
+  var
+    i,
+    offset = 4 + parseAdaptionField(packet);
+
+  if (offset >= packet.byteLength) {
+    return null;
+  }
+
+  // check for the ADTS header syncword 0xFFF
+  for (i = offset; i < packet.length - 1; i++) {
+    if (packet[i] === 0xFF && packet[i + 1] >> 4 === 0x0F) {
```

Comment:
any reason in switching to the bitshift here instead of the inverse of the check `AdtsStream` does? `(buffer[i + 1] & 0xF6) !== 0xF0`

Reply:
I just felt it was easier to read. Do you have a preference?

Reply:
I am leaning towards reverting back to the original check. From https://wiki.multimedia.cx/index.php?title=ADTS the first two bytes of the ADTS header have the format `AAAAAAAA AAAABCCD`

>B | 1 | MPEG Version: 0 for MPEG-4, 1 for MPEG-2
>C | 2 | Layer: always 0
>D | 1 | protection absent, Warning, set to 1 if there is no CRC and 0 if there is CRC

The `0xF6` is making sure that those two `C` bits are `0`, where as your check ignores those bits. I think it will be safer to make sure these bits are checked as well.